activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskAdminRuntimeImpl.java

```diff
@@ -132,4 +138,136 @@ public Task release(ReleaseTaskPayload releaseTaskPayload) {
         taskService.unclaim(releaseTaskPayload.getTaskId());
         return task(releaseTaskPayload.getTaskId());
     }
+    
+    @Override
+    public Task assign(AssignTaskPayload assignTaskPayload) {
+        // Validate that the task is visible by the currently authorized user
+        Task task;
+        try {
+            task = task(assignTaskPayload.getTaskId());
```

Comment:
@CTI777 we need to change this.. here you are an admin already.. so you can do it even if the task is not assigned to the admin.

Reply:
I simply check if task is present here... Should I skip this check?
 public Task task(String taskId) {
 org.activiti.engine.task.Task internalTask = taskService.createTaskQuery().taskId(taskId).singleResult();
        if (internalTask == null) {
            throw new NotFoundException("Unable to find task for the given id: " + taskId);
        }
        return taskConverter.from(internalTask);
    }